[BUG] json_test.py::test_structs_to_json_timestamp/arrays_to_json/maps_to_json fails

**Describe the bug**

Nightly integration tests on Spark 3.4.1 with scala 2.12 are failing in the non_utc-dev build. Looks like part of hte plan is coming out not columnar.

also saw this failing in build: 4.0 scala 2.13 cuda12 integration nightly

```
  pyspark.errors.exceptions.captured.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec

[2025-07-23T10:25:45.809Z] E               Project [to_json((timeZone,Etc/UTC), (ignoreNullFields,false), my_map#273247, Some(America/New_York)) AS mm#273249]

[2025-07-23T10:25:45.809Z] E               +- Scan ExistingRDD[my_map#273247]
```
Example of the failures:

```
[2025-07-23T10:25:46.072Z] FAILED ../../src/main/python/json_test.py::test_structs_to_json[UTC-True-Byte][DATAGEN_SEED=1753253227, TZ=America/New_York, INJECT_OOM] - pyspark.errors.exceptions.captured.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec

[2025-07-23T10:25:46.072Z] Project [to_json((timeZone,UTC), (timestampFormat,yyyy-MM-dd'T'HH:mm:ss[.SSS][XXX]), (ignoreNullFields,true), my_struct#271967, Some(America/New_York)) AS ms#271969]

[2025-07-23T10:25:46.072Z] +- Scan ExistingRDD[my_struct#271967]
```

Spark 4.0 scala 2.13 cuda12 build also had failures:

```
[Matrix of daily cuda12 combinations - scala2.13 / Matrix - SPARK_VER = '4.0.0' / Regular Test / src.main.python.json_test.test_arrays_to_json[UTC-True-Timestamp][DATAGEN_SEED=1753254352, TZ=UTC]](https://prod.blsm.nvidia.com/sw-gpu-spark-jenkins/job/rapids_integration-scala213-dev-github/440/testReport/junit/src.main.python/json_test/Matrix_of_daily_cuda12_combinations___scala2_13___Matrix___SPARK_VER____4_0_0____Regular_Test___test_arrays_to_json_UTC_True_Timestamp__DATAGEN_SEED_1753254352__TZ_UTC_/)
[Matrix of daily cuda12 combinations - scala2.13 / Matrix - SPARK_VER = '4.0.0' / Regular Test / src.main.python.json_test.test_arrays_to_json[UTC-False-Timestamp][DATAGEN_SEED=1753254352, TZ=UTC, INJECT_OOM]](https://prod.blsm.nvidia.com/sw-gpu-spark-jenkins/job/rapids_integration-scala213-dev-github/440/testReport/junit/src.main.python/json_test/Matrix_of_daily_cuda12_combinations___scala2_13___Matrix___SPARK_VER____4_0_0____Regular_Test___test_arrays_to_json_UTC_False_Timestamp__DATAGEN_SEED_1753254352__TZ_UTC__INJECT_OOM_/)
[Matrix of daily cuda12 combinations - scala2.13 / Matrix - SPARK_VER = '4.0.0' / Regular Test / src.main.python.json_test.test_arrays_to_json[Etc/UTC-True-Timestamp][DATAGEN_SEED=1753254352, TZ=UTC]](https://prod.blsm.nvidia.com/sw-gpu-spark-jenkins/job/rapids_integration-scala213-dev-github/440/testReport/junit/src.main.python/json_test/Matrix_of_daily_cuda12_combinations___scala2_13___Matrix___SPARK_VER____4_0_0____Regular_Test___test_arrays_to_json_Etc_UTC_True_Timestamp__DATAGEN_SEED_1753254352__TZ_UTC_/)
[Matrix of daily cuda12 combinations - scala2.13 / Matrix - SPARK_VER = '4.0.0' / Regular Test / src.main.python.json_test.test_arrays_to_json[Etc/UTC-False-Timestamp][DATAGEN_SEED=1753254352, TZ=UTC]](https://prod.blsm.nvidia.com/sw-gpu-spark-jenkins/job/rapids_integration-scala213-dev-github/440/testReport/junit/src.main.python/json_test/Matrix_of_daily_cuda12_combinations___scala2_13___Matrix___SPARK_VER____4_0_0____Regular_Test___test_arrays_to_json_Etc_UTC_False_Timestamp__DATAGEN_SEED_1753254352__TZ_UTC_/)
[Matrix of daily cuda12 combinations - scala2.13 / Matrix - SPARK_VER = '4.0.0' / Regular Test / src.main.python.json_test.test_maps_to_json[UTC-True-Timestamp][DATAGEN_SEED=1753254352, TZ=UTC, INJECT_OOM]](https://prod.blsm.nvidia.com/sw-gpu-spark-jenkins/job/rapids_integration-scala213-dev-github/440/testReport/junit/src.main.python/json_test/Matrix_of_daily_cuda12_combinations___scala2_13___Matrix___SPARK_VER____4_0_0____Regular_Test___test_maps_to_json_UTC_True_Timestamp__DATAGEN_SEED_1753254352__TZ_UTC__INJECT_OOM_/)
[Matrix of daily cuda12 combinations - scala2.13 / Matrix - SPARK_VER = '4.0.0' / Regular Test / src.main.python.json_test.test_maps_to_json[UTC-False-Timestamp][DATAGEN_SEED=1753254352, TZ=UTC, INJECT_OOM]](https://prod.blsm.nvidia.com/sw-gpu-spark-jenkins/job/rapids_integration-scala213-dev-github/440/testReport/junit/src.main.python/json_test/Matrix_of_daily_cuda12_combinations___scala2_13___Matrix___SPARK_VER____4_0_0____Regular_Test___test_maps_to_json_UTC_False_Timestamp__DATAGEN_SEED_1753254352__TZ_UTC__INJECT_OOM_/)
[Matrix of daily cuda12 combinations - scala2.13 / Matrix - SPARK_VER = '4.0.0' / Regular Test / src.main.python.json_test.test_maps_to_json[Etc/UTC-True-Timestamp][DATAGEN_SEED=1753254352, TZ=UTC, INJECT_OOM]](https://prod.blsm.nvidia.com/sw-gpu-spark-jenkins/job/rapids_integration-scala213-dev-github/440/testReport/junit/src.main.python/json_test/Matrix_of_daily_cuda12_combinations___scala2_13___Matrix___SPARK_VER____4_0_0____Regular_Test___test_maps_to_json_Etc_UTC_True_Timestamp__DATAGEN_SEED_1753254352__TZ_UTC__INJECT_OOM_/)
[Matrix of daily cuda12 combinations - scala2.13 / Matrix - SPARK_VER = '4.0.0' / Regular Test / src.main.python.json_test.test_maps_to_json[Etc/UTC-False-Timestamp][DATAGEN_SEED=1753254352, TZ=UTC, INJECT_OOM]](https://prod.blsm.nvidia.com/sw-gpu-spark-jenkins/job/rapids_integration-scala213-dev-github/440/testReport/junit/src.main.python/json_test/Matrix_of_daily_cuda12_combinations___scala2_13___Matrix___SPARK_VER____4_0_0____Regular_Test___test_maps_to_json_Etc_UTC_False_Timestamp__DATAGEN_SEED_1753254352__TZ_UTC__INJECT_OOM_/)
```

--- a/integration_tests/src/main/python/json_test.py
+++ b/integration_tests/src/main/python/json_test.py
@@ -1120,6 +1120,7 @@ _to_json_datagens=[byte_gen,
     'UTC',
     'Etc/UTC'
 ])
+@allow_non_gpu(*non_utc_project_allow)
 def test_structs_to_json(spark_tmp_path, data_gen, ignore_null_fields, timezone):
     struct_gen = StructGen([
         ('a', data_gen),
@@ -1152,6 +1153,7 @@ def test_structs_to_json(spark_tmp_path, data_gen, ignore_null_fields, timezone)
     'UTC',
     'Etc/UTC'
 ])
+@allow_non_gpu(*non_utc_project_allow)
 def test_arrays_to_json(spark_tmp_path, data_gen, ignore_null_fields, timezone):
     array_gen = ArrayGen(data_gen, nullable=True)
     gen = StructGen([("my_array", array_gen)], nullable=False)
@@ -1177,6 +1179,7 @@ def test_arrays_to_json(spark_tmp_path, data_gen, ignore_null_fields, timezone):
     'UTC',
     'Etc/UTC'
 ])
+@allow_non_gpu(*non_utc_project_allow)
 def test_maps_to_json(spark_tmp_path, data_gen, ignore_null_fields, timezone):
     map_gen = MapGen(StringGen('[A-Z]{1,10}', nullable=False), data_gen, nullable=True)
     gen = StructGen([("my_map", map_gen)], nullable=False)
@@ -1204,6 +1207,7 @@ def test_maps_to_json(spark_tmp_path, data_gen, ignore_null_fields, timezone):
     'UTC',
     'Etc/UTC'
 ])
+@allow_non_gpu(*non_utc_project_allow)
 def test_structs_to_json_timestamp(spark_tmp_path, data_gen, timestamp_format, timezone):
     struct_gen = StructGen([
         ("b", StructGen([('child', data_gen)], nullable=True)),

